test/widgets/content_test.dart

```diff
@@ -25,6 +28,55 @@ import 'dialog_checks.dart';
 import 'message_list_checks.dart';
 import 'page_checks.dart';
 
+/// Simulate a nested "inner" span's style by merging all ancestor-span
+/// styles, starting from the root.
+///
+/// [isInnerSpan] must return true for some descendant of [outerSpan].
+///
+/// This isn't how the style actually gets applied in the code under test
+/// (because that happens inside the engine, in SkParagraph),
+/// but it should hopefully simulate it closely enough.
+TextStyle? mergeSpanStylesOuterToInner(
+  InlineSpan outerSpan,
+  bool Function(InlineSpan) isInnerSpan,
+) {
+  final styles = <TextStyle?>[];
+  bool recurse(InlineSpan span) {
+    if (isInnerSpan(span)) {
+      styles.add(span.style);
+      return false;
+    }
+    final notInSubtree = span.visitDirectChildren(recurse);
+    if (!notInSubtree) {
+      styles.add(span.style);
+    }
+    return notInSubtree;
+  }
+  final recurseResult = recurse(outerSpan);
+  check(recurseResult).isFalse(); // check inner span was actually found
+
+  return styles.reversed.reduce((value, element) => switch ((value, element)) {
+    (TextStyle(), TextStyle()) => value!.merge(element!),
+    (TextStyle(), null)        => value,
+    (null, TextStyle())        => element,
+    (null, null)               => null,
+  });
+}
+
+/// The "merged style" ([mergeSpanStylesOuterToInner]) of a nested span.
+TextStyle? mergedStyleOfSubstring(InlineSpan rootSpan, Pattern substringPattern) {
+  return mergeSpanStylesOuterToInner(rootSpan,
+    (span) {
+      if (span is! TextSpan) return false;
+      final text = span.text;
+      if (text == null) return false;
+      return switch (substringPattern) {
+        String() => text == substringPattern,
+        _ => substringPattern.allMatches(text).isNotEmpty,
```

Comment:
These two cases don't agree in behavior — a string as a Pattern means "has this as a substring", not "equals this".

It looks like this revision doesn't end up actually using the more general Pattern case. So I think the cleanest thing is to go back to having the parameter be `String substring`, as in the previous revision.

Reply:
This revision uses the more general `Pattern` case to find text matching the RegExp we use to recognize a rendered global time:

```dart
RegExp(r'^(Tue, Jan 30|Wed, Jan 31), 2024, \d+:\d\d [AP]M$')
```

Reply:
Maybe one option is to make the cases agree in behavior, by only considering matches that cover the whole string?

```dart
      return switch (substringPattern) {
        String() => text == substringPattern,
        _ => substringPattern.allMatches(text)
          .any((match) => match.start == 0 && match.end == text.length),
      };
```

Reply:
Hmm, I see, yeah.

That solution LGTM.